addons/l10n_it_edi_extended/models/account_tax.py

```diff
@@ -0,0 +1,107 @@
+    ('TC02', 'Pension fund for accountants with a degree'),
+    ('TC03', 'Pension fund for surveyors'),
+    ('TC04', 'National pension fund for associated engineers and architects'),
+    ('TC05', 'National pension fund for notaries'),
+    ('TC06', 'Pension fund for accountants without a degree and commercial experts'),
+    ('TC07', 'ENASARCO pension fund for sales agents'),
+    ('TC08', 'ENPACL pension fund for labor consultants'),
+    ('TC09', 'ENPAM pension fund for doctors'),
+    ('TC10', 'ENPAF pension fund for chemists'),
+    ('TC11', 'ENPAV pension fund for veterinaries'),
+    ('TC12', 'ENPAIA pension fund for people working in agriculture'),
+    ('TC13', 'Pension fund for employees in delivery and marine agencies'),
+    ('TC14', 'INPGI pension fund for journalists'),
+    ('TC15', 'ONAOSI fund for sanitary orphans'),
+    ('TC16', 'CASAGIT Additional pension fund for journalists'),
+    ('TC17', 'EPPI pension fund for industrial experts'),
+    ('TC18', 'EPAP pension fund'),
+    ('TC19', 'ENPAB national pension fund for biologists'),
+    ('TC20', 'ENPAPI national pension fund for nurses'),
+    ('TC21', 'ENPAP national pension fund for psychologists'),
+    ('TC22', 'INPS national pension fund'),
+]
+
+
+class AccountTaxTemplate(models.Model):
+    _inherit = 'account.tax.template'
+
+    l10n_it_withholding_type = fields.Selection(WITHHOLDING_TYPE_SELECTION, string="Withholding tax type (Italy)", help="Withholding tax type. Only for Italian accounting EDI.")
+    l10n_it_withholding_reason = fields.Selection(WITHHOLDING_REASON_SELECTION, string="Withholding tax reason (Italy)", help="Withholding tax reason. Only for Italian accounting EDI.")
+    l10n_it_pension_fund_type = fields.Selection(PENSION_FUND_TYPE_SELECTION, string="Pension fund type (Italy)", help="Pension Fund Type. Only for Italian accounting EDI.")
+
+    def _get_tax_vals(self, company, tax_template_to_tax):
+        values = super()._get_tax_vals(company, tax_template_to_tax)
+        values.update({
+            "l10n_it_withholding_type": self.l10n_it_withholding_type,
+            "l10n_it_withholding_reason": self.l10n_it_withholding_reason,
+            "l10n_it_pension_fund_type": self.l10n_it_pension_fund_type,
+        })
+        return values
+
+class AccountTax(models.Model):
+    _inherit = 'account.tax'
+
+    l10n_it_withholding_type = fields.Selection(WITHHOLDING_TYPE_SELECTION, string="Withholding tax type (Italy)", help="Withholding tax type. Only for Italian accounting EDI.")
+    l10n_it_withholding_reason = fields.Selection(WITHHOLDING_REASON_SELECTION, string="Withholding tax reason (Italy)", help="Withholding tax reason. Only for Italian accounting EDI.")
+    l10n_it_pension_fund_type = fields.Selection(PENSION_FUND_TYPE_SELECTION, string="Pension fund type (Italy)", help="Pension Fund Type. Only for Italian accounting EDI.")
+
+    @api.constrains('l10n_it_withholding_type', 'l10n_it_withholding_reason')
```

Comment:
not quite complete, at the moment you can:
1. create negative `amount` tax, with withholding type, reason etc.
2. adjust amount to be positive, with no validation error

Reply:
O.o